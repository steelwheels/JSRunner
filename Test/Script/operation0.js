/*
 * operation0.js
 */

function main(args)
{
	let opfile = null ;
	if(isArray(args) && args.length == 1) {
		opfile = args[0] ;
		console.log("input script: " + opfile + "\n") ;
	} else {
		console.error("Invalid parameter\n") ;
		console.dump(args) ;
		return false
	}

	let url0 = URL(opfile) ;
	let op0  = Operation([url0], console) ;
	if(!checkVariables("check op", op0)){
		return false ;
	}

	let queue0 = OperationQueue() ;
	if(!queue0.execute(op0)){
		console.log("[Error] Could not execute op0\n") ;
		return false ;
	}
	queue0.waitOperations() ;

	return true ;
}
