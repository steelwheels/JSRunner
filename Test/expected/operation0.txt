Define operation object
* Hello from constructor
End of operation object
* Hello from main
